src/cmd/flash.js

```diff
@@ -55,11 +59,109 @@ module.exports = class FlashCommand extends CLICommandBase {
 		} else if (local) {
 			let allFiles = binary ? [binary, ...files] : files;
 			await this.flashLocal({ files: allFiles, applicationOnly, target });
+		} else if (tachyon) {
+			let allFiles = binary ? [binary, ...files] : files;
+			await this.flashTachyon({ verbose, files: allFiles });
 		} else {
 			await this.flashCloud({ device, files, target });
 		}
 	}
 
+	async flashTachyon({ verbose, files }) {
+		this.ui.write(`Ensure only one device is connected to a computer${os.EOL}`);
+
+		let unpackToolFolder;
+		if (files.length === 0) {
+			// If no files are passed, assume the current directory
+			unpackToolFolder = process.cwd();
+			files = await fs.readdir(unpackToolFolder);
+		} else if (files.length === 1) {
+			// If only one file is passed, check if it's a directory
+			const stats = await fs.stat(files[0]);
+			if (stats.isDirectory()) {
+				unpackToolFolder = files[0];
+				files = await fs.readdir(files[0]);
+			}
+		} else {
+			// If multiple files are passed, check the directory from the first file
+			unpackToolFolder = path.dirname(files[0]);
+		}
+
+		const parsedFiles = await this._analyzeFiles(files);
+		let linxuFiles = await this._findLinuxExecutableFiles(parsedFiles.files, { directory: unpackToolFolder });
+		linxuFiles = linxuFiles.map(f => path.basename(f));
+
+
+		const elfFiles = linxuFiles.filter(f => f.startsWith('prog') && f.endsWith('.elf'));
+		const rawProgramFiles = linxuFiles.filter(f => f.startsWith('rawprogram') && f.endsWith('.xml'));
+		const patchFiles = linxuFiles.filter(f => f.startsWith('patch') && f.endsWith('.xml'));
+
+		if (!elfFiles.length || !rawProgramFiles.length || !patchFiles.length) {
+			throw new Error('The directory should contain at least one .elf file, one rawprogram file, and one patch file');
+		}
+
+		const sortByNumber = (a, b) => {
+			const extractNumber = str => parseInt(str.match(/(\d+).xml/)[1]);
+			return extractNumber(a) - extractNumber(b);
+		};
+
+		rawProgramFiles.sort(sortByNumber);
+		patchFiles.sort(sortByNumber);
+
+		if (rawProgramFiles.length !== patchFiles.length) {
```

Comment:
I don't think this is necessary either

Reply:
I really think this is extra. What if a program doesn't have a patch file to go with it?